In [1]:
import datetime
import requests
import rdflib
start = datetime.datetime.utcnow()
LDP = rdflib.Namespace('http://www.w3.org/ns/ldp#')
print(start)

2019-12-16 18:12:37.570630


In [2]:
with open("../../sinopia_acl/.cognitoToken") as fo:
    cognitoToken = fo.read()

headers = { 
    "Authorization": f"Bearer {cognitoToken}"
}

## 2019-12-16 18:12:37.570630
Issue #[1987](https://github.com/LD4P/sinopia_editor/issues/1987), delete `harvard:RT:bf2:RareMat:Instance` from Stage.

In [3]:
result = requests.delete("https://trellis.stage.sinopia.io/repository/ld4p/harvard:RT:bf2:RareMat:Instance", headers=headers)
print(result)

<Response [204]>


## 2019-11-22 18:12:30.541639

In [4]:
result = requests.delete("https://trellis.stage.sinopia.io/repository/ld4p/UCSD:RT:bf2:Monograph:Item_OpenAccess", headers=headers)
print(result)

<Response [204]>


## 2019-11-13 19:03:12.898402

In [4]:
result = requests.delete("http://localhost:8080/repository/NeatoResourcesGroup", headers=headers)

In [7]:
print(result)

<Response [204]>


## 2019-11-08 17:30:16.013221
For production and stage, removing the following ticket https://github.com/LD4P/sinopia_editor/issues/1708

In [7]:
production_rts = """WAU:RT:RDF:list
WAU:RT:RDA:Timespan:monograph
WAU:RT:RDA:Item:etd
WAU:RT:MADSRDF:simpleSubject
WAU:RT:MADSRDF:complexSubject""".split()
for rt in production_rts:
    prod_url = f"https://trellis.sinopia.io/repository/ld4p/{rt}"
    get_result = requests.delete(prod_url, headers=headers)
    print(f"{prod_url}: result = {get_result.status_code}")

https://trellis.sinopia.io/repository/ld4p/WAU:RT:RDF:list: result = 204
https://trellis.sinopia.io/repository/ld4p/WAU:RT:RDA:Timespan:monograph: result = 204
https://trellis.sinopia.io/repository/ld4p/WAU:RT:RDA:Item:etd: result = 204
https://trellis.sinopia.io/repository/ld4p/WAU:RT:MADSRDF:simpleSubject: result = 204
https://trellis.sinopia.io/repository/ld4p/WAU:RT:MADSRDF:complexSubject: result = 204


In [11]:
stage_rts = """WAU:RT:RDF:list
WAU:RT:RDA:Timespan:monograph
WAU:RT:RDA:Item:etd
WAU:RT:MADSRDF:simpleSubject
WAU:RT:MADSRDF:complexSubject
WAU:RT:BF2:BFLC:AdminMetadata:Status:monograph:monograph
WAU:RT:BF2:BFLC:AdminMetadata:Status:monograph
WAU:RT:BF2:BFLC:AdminMetadata:monograph
""".split()
for rt in stage_rts:
    stage_url = f"https://trellis.stage.sinopia.io/repository/ld4p/{rt}"
    get_result = requests.delete(stage_url, headers=headers)
    print(f"{stage_url}: result = {get_result.status_code}")
for rt in stage_rts:
    stage_url = f"https://trellis.stage.sinopia.io/repository/ld4p/{rt}"
    get_result = requests.get(stage_url, headers=headers)
    print(f"{stage_url}: result = {get_result.status_code}")

https://trellis.stage.sinopia.io/repository/ld4p/WAU:RT:RDF:list: result = 204
https://trellis.stage.sinopia.io/repository/ld4p/WAU:RT:RDA:Timespan:monograph: result = 204
https://trellis.stage.sinopia.io/repository/ld4p/WAU:RT:RDA:Item:etd: result = 204
https://trellis.stage.sinopia.io/repository/ld4p/WAU:RT:MADSRDF:simpleSubject: result = 204
https://trellis.stage.sinopia.io/repository/ld4p/WAU:RT:MADSRDF:complexSubject: result = 204
https://trellis.stage.sinopia.io/repository/ld4p/WAU:RT:BF2:BFLC:AdminMetadata:Status:monograph:monograph: result = 204
https://trellis.stage.sinopia.io/repository/ld4p/WAU:RT:BF2:BFLC:AdminMetadata:Status:monograph: result = 204
https://trellis.stage.sinopia.io/repository/ld4p/WAU:RT:BF2:BFLC:AdminMetadata:monograph: result = 204
https://trellis.stage.sinopia.io/repository/ld4p/WAU:RT:RDF:list: result = 410
https://trellis.stage.sinopia.io/repository/ld4p/WAU:RT:RDA:Timespan:monograph: result = 410
https://trellis.stage.sinopia.io/repository/ld4p/WAU:RT

## 2019-11-05 00:40:56.621468

In [4]:
rt_to_delete = """UCSD:RT:BF2:Identifiers:ISBN
UCSD:RT:BF2:Identifiers:ISMN
UCSD:RT:BF2:Identifiers:LCCN
UCSD:RT:BF2:Identifiers:Local
UCSD:RT:BF2:Identifiers:Distributor
UCSD:RT:BF2:Identifiers:MusicPlate
UCSD:RT:BF2:Identifiers:MusicPub
UCSD:RT:BF2:Identifiers:Other
UCSD:RT:BF2:Item:Use_OpenAccess
UCSDld4p:RT:bf2:Monograph:Item""".split()

In [8]:
for rt in rt_to_delete:
    stage_url = f"https://trellis.stage.sinopia.io/repository/ld4p/{rt}"
    get_result = requests.get(stage_url, headers=headers)
    print(f"{stage_url}: result = {get_result.status_code}")

https://trellis.stage.sinopia.io/repository/ld4p/UCSD:RT:BF2:Identifiers:ISBN: result = 410
https://trellis.stage.sinopia.io/repository/ld4p/UCSD:RT:BF2:Identifiers:ISMN: result = 410
https://trellis.stage.sinopia.io/repository/ld4p/UCSD:RT:BF2:Identifiers:LCCN: result = 410
https://trellis.stage.sinopia.io/repository/ld4p/UCSD:RT:BF2:Identifiers:Local: result = 410
https://trellis.stage.sinopia.io/repository/ld4p/UCSD:RT:BF2:Identifiers:Distributor: result = 410
https://trellis.stage.sinopia.io/repository/ld4p/UCSD:RT:BF2:Identifiers:MusicPlate: result = 410
https://trellis.stage.sinopia.io/repository/ld4p/UCSD:RT:BF2:Identifiers:MusicPub: result = 410
https://trellis.stage.sinopia.io/repository/ld4p/UCSD:RT:BF2:Identifiers:Other: result = 410
https://trellis.stage.sinopia.io/repository/ld4p/UCSD:RT:BF2:Item:Use_OpenAccess: result = 410
https://trellis.stage.sinopia.io/repository/ld4p/UCSDld4p:RT:bf2:Monograph:Item: result = 410


## 2019-10-30 18:04:08.268981


In [3]:
result = requests.delete('https://trellis.sinopia.io/repository/penn/f6ca9161-2a30-4836-85e7-a291065d57cd', headers=headers)

In [6]:
print(result.status_code)

204


## 2019-10-09 15:53:42.364304

In [3]:
# Removing local copy of Title RT
result = requests.delete("http://localhost:8080/repository/ld4p/jpn:work:title",
                         headers=headers)

In [4]:
print(result)

<Response [204]>


## 2019-09-18 18:48:39.481603


In [3]:
result = requests.delete("https://trellis.stage.sinopia.io/repository/ld4p/UChicago:RT:bf2:RareMat:Item",
                         headers=headers)
print(result)

<Response [204]>


In [4]:
result2 = requests.delete("https://trellis.stage.sinopia.io/repository/ld4p/UCHicago:RT:bf2:RareMat:Item",
                         headers=headers)
print(result2)

<Response [204]>


## 2019-09-11 17:57:23.418161

In [45]:
result = requests.delete("https://trellis.stage.sinopia.io/repository/ld4p/WAU:RT:RDA:Manifestation:monograph",
                         headers=headers)

In [46]:
print(result)

<Response [204]>


##  2019-09-10 14:33:08.782954
# Sinopia Stage and Production RT and Resources deletions
This is for issue [1295](https://github.com/LD4P/sinopia_editor/issues/1295)

In [6]:
keep_rt = "UCSD:RT:BF2:RightsModuleTest:Work:CcLicenseModel"

In [35]:
def reload_ld4p(group_url = f"https://trellis.stage.sinopia.io/repository/ld4p"):
    ld4p_group = requests.get(group_url)
    ld4p_graph = rdflib.ConjunctiveGraph()
    ld4p_graph.parse(data=ld4p_group.text, format='turtle')
    return ld4p_graph

In [19]:
ld4p_graph = reload_ld4p("stage")


In [ ]:
rts = []
for obj in ld4p_graph.objects(predicate=LDP.contains):
    rt_url = str(obj)
    rt_id = rt_url.split("/")[-1]
    if rt_id.startswith("sinopia") or rt_id == keep_rt:
        print(str(obj))
    else:
        rts.append(str(obj))

In [22]:
len(rts)

822

In [24]:
print(rts[3])

https://trellis.stage.sinopia.io/repository/ld4p/hrc:rt:arm:bibliographicCitation:page


In [25]:
for row in rts:
    delete_result = requests.delete(row, headers=headers)
    print(f"{row} {delete_result.status_code}")

https://trellis.stage.sinopia.io/repository/ld4p/harvard:test2:resourceTemplate:bf2:FileType 204
https://trellis.stage.sinopia.io/repository/ld4p/hrc:rt:custodialHistory:buyer 204
https://trellis.stage.sinopia.io/repository/ld4p/uChicago:RT:RDA:MANIFESTATION:MONOGRAPH 204
https://trellis.stage.sinopia.io/repository/ld4p/hrc:rt:arm:bibliographicCitation:page 204
https://trellis.stage.sinopia.io/repository/ld4p/stanford:RT:BF2:ProvisionPlace 204
https://trellis.stage.sinopia.io/repository/ld4p/Yale:RT:bf2:Monograph:Item 204
https://trellis.stage.sinopia.io/repository/ld4p/UNAM:RT:BF2:Acceso 204
https://trellis.stage.sinopia.io/repository/ld4p/PMO:RT:SoundRecording:Work 204
https://trellis.stage.sinopia.io/repository/ld4p/harvard:test2:resourceTemplate:bf2:Monograph:Item 204
https://trellis.stage.sinopia.io/repository/ld4p/WAU:RT:RDA:Expression:Monograph 204
https://trellis.stage.sinopia.io/repository/ld4p/harvard:test2:resourceTemplate:bf2:TypeRec 204
https://trellis.stage.sinopia.io/rep

In [32]:
def delete_trellis_rdf(trellis_url = 'https://trellis.stage.sinopia.io/repository'):
    start = datetime.datetime.utcnow()
    print(f"Starting Deletion of {trellis_url} at {start}")
    home = requests.get(trellis_url)
    home_graph = rdflib.ConjunctiveGraph()
    home_graph.parse(data=home.text, format='turtle')
    test, training = [], []
    for o in home_graph.objects(predicate=LDP.contains):
        group_result = requests.get(str(o))
        if str(o).endswith("ld4p"):
            continue
        print(f"Institution URI {o}")
        group_graph = rdflib.ConjunctiveGraph()
        group_graph.parse(data=group_result.text, format='turtle')
        for i, resource_uri in enumerate(
            group_graph.objects(predicate=LDP.contains)):
            delete_result = requests.get(str(resource_uri), headers=headers)
            if delete_result.status_code > 300:
                print(f"Error deleting {resource_iri}, status code {delete_result.status_code}")
        print(f"Finished with {o}")
    end = datetime.datetime.utcnow()
    print(f"""Finished deleting at {end}, total time: {(end-start).seconds} seconds""")

In [33]:
delete_trellis_rdf()

Starting Deletion of https://trellis.stage.sinopia.io/repository at 2019-09-10 16:47:40.284357
Institution URI https://trellis.stage.sinopia.io/repository/stanford
Finished with https://trellis.stage.sinopia.io/repository/stanford
Institution URI https://trellis.stage.sinopia.io/repository/frick
Finished with https://trellis.stage.sinopia.io/repository/frick
Institution URI https://trellis.stage.sinopia.io/repository/ucdavis
Finished with https://trellis.stage.sinopia.io/repository/ucdavis
Institution URI https://trellis.stage.sinopia.io/repository/pcc
Finished with https://trellis.stage.sinopia.io/repository/pcc
Institution URI https://trellis.stage.sinopia.io/repository/northwestern
Finished with https://trellis.stage.sinopia.io/repository/northwestern
Institution URI https://trellis.stage.sinopia.io/repository/princeton
Finished with https://trellis.stage.sinopia.io/repository/princeton
Institution URI https://trellis.stage.sinopia.io/repository/minnesota
Finished with https://trell

In [36]:
prod_graph = reload_ld4p(group_url = "https://trellis.sinopia.io/repository/ld4p")


In [37]:
rts = []
for obj in prod_graph.objects(predicate=LDP.contains):
    rt_url = str(obj)
    rt_id = rt_url.split("/")[-1]
    if rt_id.startswith("sinopia") or rt_id == keep_rt:
        print(str(obj))
    else:
        rts.append(str(obj))

https://trellis.sinopia.io/repository/ld4p/sinopia:resourceTemplate:bf2:Item:ImmAcqSource
https://trellis.sinopia.io/repository/ld4p/sinopia:resourceTemplate:bf2:LCC
https://trellis.sinopia.io/repository/ld4p/sinopia:resourceTemplate:bf2:Identifiers:DDC
https://trellis.sinopia.io/repository/ld4p/sinopia:resourceTemplate:bf2:Identifiers:Barcode
https://trellis.sinopia.io/repository/ld4p/sinopia:resourceTemplate:bf2:Title
https://trellis.sinopia.io/repository/ld4p/sinopia:resourceTemplate:bf2:FormTest
https://trellis.sinopia.io/repository/ld4p/sinopia:resourceTemplate:bf2:SupplContent
https://trellis.sinopia.io/repository/ld4p/sinopia:resourceTemplate:bf2:TopicInput
https://trellis.sinopia.io/repository/ld4p/sinopia:resourceTemplate:bf2:Item:Enumeration
https://trellis.sinopia.io/repository/ld4p/sinopia:resourceTemplate:bf2:Capture:Place
https://trellis.sinopia.io/repository/ld4p/sinopia:resourceTemplate:bf2:Topic:madsTemporal
https://trellis.sinopia.io/repository/ld4p/sinopia:resourceTe

In [39]:
rts[9]

'https://trellis.sinopia.io/repository/ld4p/UNAM:RT:BF2:Nota'

In [40]:
for row in rts:
    delete_result = requests.delete(row, headers=headers)
    print(f"{row} {delete_result.status_code}")

https://trellis.sinopia.io/repository/ld4p/profile:bf2:SysReq 204
https://trellis.sinopia.io/repository/ld4p/profile:bf2:Title:KeyTitle 204
https://trellis.sinopia.io/repository/ld4p/profile:bf2:Contents 204
https://trellis.sinopia.io/repository/ld4p/UNAM:RT:BF2:GeneroForma 204
https://trellis.sinopia.io/repository/ld4p/lc:RT:bf2:PrintPhoto:AcqSource 204
https://trellis.sinopia.io/repository/ld4p/Frick:resourceTemplate:bf2:Monograph:Instance 204
https://trellis.sinopia.io/repository/ld4p/UNAM:RT:BF2:Identificadores:Otro 204
https://trellis.sinopia.io/repository/ld4p/profile:bf2:Title:CollTitle 204
https://trellis.sinopia.io/repository/ld4p/UNAM:RT:BF2:Uso 204
https://trellis.sinopia.io/repository/ld4p/UNAM:RT:BF2:Nota 204
https://trellis.sinopia.io/repository/ld4p/UNAM:RT:BF2:Seriada:Obra 204
https://trellis.sinopia.io/repository/ld4p/UNAM:RT:BF2:Agentes:Organismo 204
https://trellis.sinopia.io/repository/ld4p/UNAM:RT:BF2:Seriada:Frecuencia 204
https://trellis.sinopia.io/repository/ld4

In [42]:
for 

124

## 2019-08-19 16:31:45.882460
# Remove a few templates from DEV, those starting with lc:RT https://github.com/LD4P/sinopia_editor/issues/1255

In [68]:
to_be_deleted = ["lc:RT:bf2:AdminMetadata:BFDB", 
                 "lc:RT:bf2:AdminMetadata",
                 "lc:RT:bf2:AdminMetadata:GenerationProcess",
                 "lc:RT:bf2:ReferenceInstance",
                 "lc:RT:bf2:RelatedExpression",
                 "lc:RT:bf2:RelatedInstance",
                 "lc:RT:bf2:RelatedWork",
                 "lc:RT:bf2:Relation",
                 "lc:RT:bf2:AdminMetadata:Status"]

In [73]:
## Doublecheck that they exist
for rt in to_be_deleted:
    uri = f"https://trellis.development.sinopia.io/repository/ld4p/{rt}"
    result = requests.get(uri)
    print(uri, result.status_code)

https://trellis.development.sinopia.io/repository/ld4p/lc:RT:bf2:AdminMetadata:BFDB 200
https://trellis.development.sinopia.io/repository/ld4p/lc:RT:bf2:AdminMetadata 200
https://trellis.development.sinopia.io/repository/ld4p/lc:RT:bf2:AdminMetadata:GenerationProcess 200
https://trellis.development.sinopia.io/repository/ld4p/lc:RT:bf2:ReferenceInstance 200
https://trellis.development.sinopia.io/repository/ld4p/lc:RT:bf2:RelatedExpression 200
https://trellis.development.sinopia.io/repository/ld4p/lc:RT:bf2:RelatedInstance 200
https://trellis.development.sinopia.io/repository/ld4p/lc:RT:bf2:RelatedWork 200
https://trellis.development.sinopia.io/repository/ld4p/lc:RT:bf2:Relation 200
https://trellis.development.sinopia.io/repository/ld4p/lc:RT:bf2:AdminMetadata:Status 200


In [74]:
# Delete these rt
for rt in to_be_deleted:
    uri = f"https://trellis.development.sinopia.io/repository/ld4p/{rt}"
    delete_request = requests.delete(uri, headers=headers)
    print(uri, delete_request.status_code)

https://trellis.development.sinopia.io/repository/ld4p/lc:RT:bf2:AdminMetadata:BFDB 204
https://trellis.development.sinopia.io/repository/ld4p/lc:RT:bf2:AdminMetadata 204
https://trellis.development.sinopia.io/repository/ld4p/lc:RT:bf2:AdminMetadata:GenerationProcess 204
https://trellis.development.sinopia.io/repository/ld4p/lc:RT:bf2:ReferenceInstance 204
https://trellis.development.sinopia.io/repository/ld4p/lc:RT:bf2:RelatedExpression 204
https://trellis.development.sinopia.io/repository/ld4p/lc:RT:bf2:RelatedInstance 204
https://trellis.development.sinopia.io/repository/ld4p/lc:RT:bf2:RelatedWork 204
https://trellis.development.sinopia.io/repository/ld4p/lc:RT:bf2:Relation 204
https://trellis.development.sinopia.io/repository/ld4p/lc:RT:bf2:AdminMetadata:Status 204


In [75]:
# Print-out for Issue Comment
for rt in to_be_deleted:
    print(f"- {rt}")

- lc:RT:bf2:AdminMetadata:BFDB
- lc:RT:bf2:AdminMetadata
- lc:RT:bf2:AdminMetadata:GenerationProcess
- lc:RT:bf2:ReferenceInstance
- lc:RT:bf2:RelatedExpression
- lc:RT:bf2:RelatedInstance
- lc:RT:bf2:RelatedWork
- lc:RT:bf2:Relation
- lc:RT:bf2:AdminMetadata:Status


## 2019-08-15 22:57:52.944649 for Sinopia 
# Issue wipe RTs on DEV https://github.com/LD4P/sinopia_editor/issues/1243

In [35]:
ld4p_group = requests.get('https://trellis.development.sinopia.io/repository/ld4p')
ld4p_graph = rdflib.ConjunctiveGraph()
ld4p_graph.parse(data=ld4p_group.text, format='turtle')

<Graph identifier=Na97f4a1dda7c427fb45e6f520cc520fb (<class 'rdflib.graph.Graph'>)>

In [20]:
LDP = rdflib.Namespace('http://www.w3.org/ns/ldp#')

In [32]:
print(ld4p_graph.serialize(format='turtle').decode())

@prefix acl: <http://www.w3.org/ns/auth/acl#> .
@prefix as: <https://www.w3.org/ns/activitystreams#> .
@prefix dc: <http://purl.org/dc/terms/> .
@prefix dc11: <http://purl.org/dc/elements/1.1/> .
@prefix geo: <http://www.w3.org/2003/01/geo/wgs84_pos#> .
@prefix ldp: <http://www.w3.org/ns/ldp#> .
@prefix memento: <http://mementoweb.org/ns#> .
@prefix owl: <http://www.w3.org/2002/07/owl#> .
@prefix prov: <http://www.w3.org/ns/prov#> .
@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
@prefix schema: <http://schema.org/> .
@prefix skos: <http://www.w3.org/2004/02/skos/core#> .
@prefix time: <http://www.w3.org/2006/time#> .
@prefix xml: <http://www.w3.org/XML/1998/namespace> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .

<https://trellis.development.sinopia.io/repository/ld4p> rdfs:label "LD4P Group",
        "Sinopia LD4P Group Container" ;
    ldp:contains <https://trellis.development.sinopia.io/repository/ld4p/Cor

In [47]:
rts = []
for obj in ld4p_graph.objects(predicate=LDP.contains):
    rts.append(str(obj))

In [48]:
len(rts)

483

In [23]:
def delete_rts(env):
    start = datetime.datetime.utcnow()
    print(f"Started at {start}")
    size = 0
    ld4p_group = requests.get(f'https://trellis.{env}.sinopia.io/repository/ld4p')
    ld4p_graph = rdflib.ConjunctiveGraph()
    ld4p_graph.parse(data=ld4p_group.text, format='turtle')
    for obj in ld4p_graph.objects(predicate=LDP.contains):
        print(obj, end="")
        delete_request = requests.delete(str(obj), headers=headers)
        print(f"\t{delete_request.status_code}")
        if delete_request.status_code > 300:
            print("Error")
        size += 1
    end = datetime.datetime.utcnow()
    print(f"Finished at {end} total time {(end-start).seconds} for {size}")
    return ld4p_graph

In [53]:
delete_rts('development')

Started at 2019-08-15 23:56:16.784669
https://trellis.development.sinopia.io/repository/ld4p/sinopia:resourceTemplate:bf2:Monograph:Instance	204
Error
https://trellis.development.sinopia.io/repository/ld4p/WAU:RT:RDA:MANIFESTATION:MONOGRAPH	204
Error
https://trellis.development.sinopia.io/repository/ld4p/sinopia:resourceTemplate:bf2:Agents:RWO:LocationOfConference	204
Error
https://trellis.development.sinopia.io/repository/ld4p/sinopia:resourceTemplate:bf2:AdminMetadata2	204
Error
https://trellis.development.sinopia.io/repository/ld4p/sinopia:resourceTemplate:bf2:Provision:PlaceTest	204
Error
https://trellis.development.sinopia.io/repository/ld4p/sinopia:resourceTemplate:bf2:Monograph:RelatedInstance	204
Error
https://trellis.development.sinopia.io/repository/ld4p/sinopia:resourceTemplate:bf2:Form	204
Error
https://trellis.development.sinopia.io/repository/ld4p/sinopia:resourceTemplate:bf2:Identifiers:Matrix	204
Error
https://trellis.development.sinopia.io/repository/ld4p/CornellSinatr

AttributeError: 'datetime.datetime' object has no attribute 'seconds'

In [57]:
delete_rts('development')

Started at 2019-08-15 23:59:59.525076
https://trellis.development.sinopia.io/repository/ld4p/istodayover?	404
Error
Finished at 2019-08-16 00:00:00.335160 total time 0 for 1


<Graph identifier=N7cff7eceead34fe5b3321dd409b5ab52 (<class 'rdflib.graph.ConjunctiveGraph'>)>

In [54]:
now = datetime.datetime.utcnow()

In [55]:
end = datetime.datetime.utcnow()
(end-now).seconds


19

In [60]:
delete_request = requests.delete("https://trellis.development.sinopia.io/repository/ld4p/istodayover%3F", headers=headers)

In [61]:
print(delete_request)

<Response [204]>


In [43]:
for obj in ld4p_graph.objects(predicate=LDP.contains):
    print(f"- Deleted {obj}")


- Deleted https://trellis.development.sinopia.io/repository/ld4p/lc:RT:bf2:Form
- Deleted https://trellis.development.sinopia.io/repository/ld4p/lc:RT:bf2:Monograph:Instance
- Deleted https://trellis.development.sinopia.io/repository/ld4p/lc:RT:bf2:Cartographic:Attributes
- Deleted https://trellis.development.sinopia.io/repository/ld4p/lc:RT:bf2:Capture:Place
- Deleted https://trellis.development.sinopia.io/repository/ld4p/middle_of_alphabet2
- Deleted https://trellis.development.sinopia.io/repository/ld4p/sinopia:resourceTemplate:bf2:Resolution
- Deleted https://trellis.development.sinopia.io/repository/ld4p/sinopia:resourceTemplate:bf2:Agents:RelatedAgents
- Deleted https://trellis.development.sinopia.io/repository/ld4p/sinopia:resourceTemplate:bf2:SoundContent
- Deleted https://trellis.development.sinopia.io/repository/ld4p/lc:RT:bf2:Serial:RelatedInstance
- Deleted https://trellis.development.sinopia.io/repository/ld4p/lc:RT:bf2:CaptureStorage
- Deleted https://trellis.development.

In [49]:
ld4p_group2 = requests.get('https://trellis.development.sinopia.io/repository/ld4p')
ld4p_graph2 = rdflib.ConjunctiveGraph()
ld4p_graph2.parse(data=ld4p_group.text, format='turtle')

<Graph identifier=N1f57adf9671541a7b0aa0122f82f1079 (<class 'rdflib.graph.Graph'>)>

In [50]:
for obj in ld4p_graph2.objects(predicate=LDP.contains):
    print(obj)

https://trellis.development.sinopia.io/repository/ld4p/lc:RT:bf2:35mmFeatureFilm:TypeRec
https://trellis.development.sinopia.io/repository/ld4p/sinopia:resourceTemplate:bf2:Identifiers:LC
https://trellis.development.sinopia.io/repository/ld4p/lc:RT:bf2:NotatedMusic:Work
https://trellis.development.sinopia.io/repository/ld4p/lc:RT:bf2:PlayingSpeed
https://trellis.development.sinopia.io/repository/ld4p/profile:bf2:SupplContent
https://trellis.development.sinopia.io/repository/ld4p/lc:RT:bf2:MIBluRayDVD:Summary
https://trellis.development.sinopia.io/repository/ld4p/lc:RT:bf2:Serial:Title
https://trellis.development.sinopia.io/repository/ld4p/lc:RT:bf2:Monograph:RelatedInstance
https://trellis.development.sinopia.io/repository/ld4p/lc:RT:bf2:Identifiers:Local
https://trellis.development.sinopia.io/repository/ld4p/lc:RT:bflc:Agents:PrimaryContribution
https://trellis.development.sinopia.io/repository/ld4p/WAU:RT:RDA:Work:monograph
https://trellis.development.sinopia.io/repository/ld4p/WAU:R

<hr>

In [6]:
home_prod = requests.get('https://trellis.sinopia.io/repository')
home_prod_graph = rdflib.ConjunctiveGraph()
home_prod_graph.parse(data=home_prod.text, format='turtle')

<Graph identifier=Ne1dec887882249a3977f864638ab6e3f (<class 'rdflib.graph.Graph'>)>

In [8]:
def get_resources(graph):
    rdf_resources = []
    for obj in graph.objects(predicate=LDP.contains):
        if str(obj).endswith("ld4p"):
            continue
        print(obj)
        group_result = requests.get(str(obj))
        group_graph = rdflib.ConjunctiveGraph()
        group_graph.parse(data=group_result.text, format='turtle')
        for resource in group_graph.objects(predicate=rdflib.URIRef('http://www.w3.org/ns/ldp#contains')):
            rdf_resources.append(resource)
            print(".", end="")
    return rdf_resources

In [10]:
rdf_prod = get_resources(home_prod_graph)

https://trellis.sinopia.io/repository/yale
.https://trellis.sinopia.io/repository/michigan
https://trellis.sinopia.io/repository/dlc
https://trellis.sinopia.io/repository/princeton
https://trellis.sinopia.io/repository/hrc
https://trellis.sinopia.io/repository/ucdavis
.https://trellis.sinopia.io/repository/harvard
https://trellis.sinopia.io/repository/penn
https://trellis.sinopia.io/repository/duke
https://trellis.sinopia.io/repository/washington
https://trellis.sinopia.io/repository/frick
https://trellis.sinopia.io/repository/ucsd
https://trellis.sinopia.io/repository/northwestern
https://trellis.sinopia.io/repository/pcc
..https://trellis.sinopia.io/repository/nlm
https://trellis.sinopia.io/repository/minnesota
https://trellis.sinopia.io/repository/chicago
https://trellis.sinopia.io/repository/stanford
https://trellis.sinopia.io/repository/cornell
...https://trellis.sinopia.io/repository/tamu
https://trellis.sinopia.io/repository/alberta
https://trellis.sinopia.io/repository/boulder


In [12]:
for row in rdf_prod:
    delete_result = requests.delete(str(row), headers=headers)
    print(row, delete_result)

https://trellis.sinopia.io/repository/yale/e3ec15a8-5143-41e5-8e0b-a7453f3418d0 <Response [204]>
https://trellis.sinopia.io/repository/ucdavis/0c9895ff-9470-4c70-bee4-b791ee179b34 <Response [204]>
https://trellis.sinopia.io/repository/pcc/a002fe6b-9f04-4cec-84a1-655942d08610 <Response [204]>
https://trellis.sinopia.io/repository/pcc/6a4d87f7-03b2-4c7e-8110-d207bf8f4e5d <Response [204]>
https://trellis.sinopia.io/repository/cornell/0db2056b-9feb-4bf1-8969-d2d2df673a46 <Response [204]>
https://trellis.sinopia.io/repository/cornell/d271da71-3db3-4ab9-98ed-f55828f4108b <Response [204]>
https://trellis.sinopia.io/repository/cornell/9f11fcc8-0572-4f4a-83b2-3c28bff1e999 <Response [204]>


In [13]:
for row in rdf_prod:
    print(f"- {row}")

- https://trellis.sinopia.io/repository/yale/e3ec15a8-5143-41e5-8e0b-a7453f3418d0
- https://trellis.sinopia.io/repository/ucdavis/0c9895ff-9470-4c70-bee4-b791ee179b34
- https://trellis.sinopia.io/repository/pcc/a002fe6b-9f04-4cec-84a1-655942d08610
- https://trellis.sinopia.io/repository/pcc/6a4d87f7-03b2-4c7e-8110-d207bf8f4e5d
- https://trellis.sinopia.io/repository/cornell/0db2056b-9feb-4bf1-8969-d2d2df673a46
- https://trellis.sinopia.io/repository/cornell/d271da71-3db3-4ab9-98ed-f55828f4108b
- https://trellis.sinopia.io/repository/cornell/9f11fcc8-0572-4f4a-83b2-3c28bff1e999


In [14]:
len(rdf_prod)

7

In [3]:
home_stage = requests.get('https://trellis.stage.sinopia.io/repository')

In [4]:
home_stage 

<Response [200]>

In [9]:
home_stage_graph = rdflib.ConjunctiveGraph()
home_stage_graph.parse(data=home_stage.text, format='turtle')

<Graph identifier=N7f0c43ad205345aabad29747659f3625 (<class 'rdflib.graph.Graph'>)>

In [26]:
rdf_resources = []
for o in home_stage_graph.objects(predicate=rdflib.URIRef('http://www.w3.org/ns/ldp#contains')):
    if str(o).endswith("ld4p"):
        continue
    print(o)
    group_result = requests.get(str(o))
    group_graph = rdflib.ConjunctiveGraph()
    group_graph.parse(data=group_result.text, format='turtle')
    for resource in group_graph.objects(predicate=rdflib.URIRef('http://www.w3.org/ns/ldp#contains')):
        rdf_resources.append(resource)
        print(resource)
    

https://trellis.stage.sinopia.io/repository/duke
https://trellis.stage.sinopia.io/repository/boulder
https://trellis.stage.sinopia.io/repository/dlc
https://trellis.stage.sinopia.io/repository/dlc/51848894-f347-4d96-9a02-f4080fea600d
https://trellis.stage.sinopia.io/repository/harvard
https://trellis.stage.sinopia.io/repository/harvard/afa61233-385f-4cf2-a07d-aa6975228d2f
https://trellis.stage.sinopia.io/repository/harvard/9719dc4a-142a-4214-818f-e45156a0040b
https://trellis.stage.sinopia.io/repository/harvard/9ac76d57-fe8d-4cbd-bc35-44b119e2b40e
https://trellis.stage.sinopia.io/repository/harvard/b477e25c-03b1-4621-8dc9-7feca2960f6d
https://trellis.stage.sinopia.io/repository/harvard/d03e3925-d768-437b-91de-cc4bd6b94934
https://trellis.stage.sinopia.io/repository/harvard/0e0f6e3c-166e-40ac-8293-19c9f2cb166f
https://trellis.stage.sinopia.io/repository/penn
https://trellis.stage.sinopia.io/repository/penn/3a189a3e-a541-4612-b276-e23d2010e7e6
https://trellis.stage.sinopia.io/repository/c

In [28]:
len(rdf_resources)

36

In [34]:
delete_one = requests.delete(str(rdf_resources[-1]), headers=headers)
print(delete_one, delete_one.text)

<Response [204]> 


In [30]:
delete_one

<Response [401]>

In [27]:
rdf_resources[-1]

rdflib.term.URIRef('https://trellis.stage.sinopia.io/repository/minnesota/767282cd-bc39-4bd8-ade0-7a8d1e0f9c27')

In [24]:
headers

{'Authorization': 'Bearer eyJraWQiOiJ5SnczaUFaOWsrRVdQekxhakVFbVwvM1FzdUZzT0hcL2NucWdyWWhcL2VsRHZvPSIsImFsZyI6IlJTMjU2In0.eyJzdWIiOiIxYzJiNDFkNS1lYzE5LTQxY2ItOWNkOC03OTA3NDFkNDhmYzEiLCJldmVudF9pZCI6ImFiMjM2NjU0LTZhZDgtNDVkNy1iMjM1LTJiMGZkZmNkYzkwZSIsInRva2VuX3VzZSI6ImFjY2VzcyIsInNjb3BlIjoiYXdzLmNvZ25pdG8uc2lnbmluLnVzZXIuYWRtaW4iLCJhdXRoX3RpbWUiOjE1NjUxOTE4NzQsImlzcyI6Imh0dHBzOlwvXC9jb2duaXRvLWlkcC51cy13ZXN0LTIuYW1hem9uYXdzLmNvbVwvdXMtd2VzdC0yX2lsTVFXME0wUiIsImV4cCI6MTU2NTE5NTQ3NCwiaWF0IjoxNTY1MTkxODc0LCJqdGkiOiJmNDM3MmU2Ni00MDFhLTRlODgtOWI3MS1jZmNmMzM2NzcyYjEiLCJjbGllbnRfaWQiOiIyM2wzMHBidTc0cHVzZ2g3Y210dWZiOGQwNSIsInVzZXJuYW1lIjoianBuZWxzb24ifQ.d69E73-NnucYRGzpMgvy9qT2lqnfYutH69qX388cJBRs-uPG7-iOlX6hEfpLE4yoSClR2tNxC-fGLtss3nA80PCe7_4aM6HlGqn2ULGkv1H6xuMhLnwYp9jr_4H997AMRkyW_gYTeZXxp54zg7eWlUxUyj8d5f7Xv96tn2aELAxgvx8wQVvd82PEXt3gE5681gGhY1AEFITWrmlfWZ-4Lpr9kV0vncVYocVLbpbzGqbUK3OYLx5O0S1cIwW0T2z3trux56rL8UlnvdxSz9mVD4prDF4GASoTx7Z0cHeFqVGgaJc7zvucOnxHJDyemcoH7A760VV93X84JgHGKjFePw'}

In [33]:
rdf_resources

[rdflib.term.URIRef('https://trellis.stage.sinopia.io/repository/dlc/51848894-f347-4d96-9a02-f4080fea600d'),
 rdflib.term.URIRef('https://trellis.stage.sinopia.io/repository/harvard/afa61233-385f-4cf2-a07d-aa6975228d2f'),
 rdflib.term.URIRef('https://trellis.stage.sinopia.io/repository/harvard/9719dc4a-142a-4214-818f-e45156a0040b'),
 rdflib.term.URIRef('https://trellis.stage.sinopia.io/repository/harvard/9ac76d57-fe8d-4cbd-bc35-44b119e2b40e'),
 rdflib.term.URIRef('https://trellis.stage.sinopia.io/repository/harvard/b477e25c-03b1-4621-8dc9-7feca2960f6d'),
 rdflib.term.URIRef('https://trellis.stage.sinopia.io/repository/harvard/d03e3925-d768-437b-91de-cc4bd6b94934'),
 rdflib.term.URIRef('https://trellis.stage.sinopia.io/repository/harvard/0e0f6e3c-166e-40ac-8293-19c9f2cb166f'),
 rdflib.term.URIRef('https://trellis.stage.sinopia.io/repository/penn/3a189a3e-a541-4612-b276-e23d2010e7e6'),
 rdflib.term.URIRef('https://trellis.stage.sinopia.io/repository/stanford/10a1f602-bc79-4271-a0f3-05c91

In [35]:
rdf_resources[-1]

rdflib.term.URIRef('https://trellis.stage.sinopia.io/repository/minnesota/767282cd-bc39-4bd8-ade0-7a8d1e0f9c27')

In [36]:
for row in rdf_resources:
    delete_result = requests.delete(str(row), headers=headers)
    print(row, delete_result)

https://trellis.stage.sinopia.io/repository/dlc/51848894-f347-4d96-9a02-f4080fea600d <Response [204]>
https://trellis.stage.sinopia.io/repository/harvard/afa61233-385f-4cf2-a07d-aa6975228d2f <Response [204]>
https://trellis.stage.sinopia.io/repository/harvard/9719dc4a-142a-4214-818f-e45156a0040b <Response [204]>
https://trellis.stage.sinopia.io/repository/harvard/9ac76d57-fe8d-4cbd-bc35-44b119e2b40e <Response [204]>
https://trellis.stage.sinopia.io/repository/harvard/b477e25c-03b1-4621-8dc9-7feca2960f6d <Response [204]>
https://trellis.stage.sinopia.io/repository/harvard/d03e3925-d768-437b-91de-cc4bd6b94934 <Response [204]>
https://trellis.stage.sinopia.io/repository/harvard/0e0f6e3c-166e-40ac-8293-19c9f2cb166f <Response [204]>
https://trellis.stage.sinopia.io/repository/penn/3a189a3e-a541-4612-b276-e23d2010e7e6 <Response [204]>
https://trellis.stage.sinopia.io/repository/stanford/10a1f602-bc79-4271-a0f3-05c91e7ea613 <Response [204]>
https://trellis.stage.sinopia.io/repository/stanford

In [38]:
for row in rdf_resources:
    print(f"- {row}")

- https://trellis.stage.sinopia.io/repository/dlc/51848894-f347-4d96-9a02-f4080fea600d
- https://trellis.stage.sinopia.io/repository/harvard/afa61233-385f-4cf2-a07d-aa6975228d2f
- https://trellis.stage.sinopia.io/repository/harvard/9719dc4a-142a-4214-818f-e45156a0040b
- https://trellis.stage.sinopia.io/repository/harvard/9ac76d57-fe8d-4cbd-bc35-44b119e2b40e
- https://trellis.stage.sinopia.io/repository/harvard/b477e25c-03b1-4621-8dc9-7feca2960f6d
- https://trellis.stage.sinopia.io/repository/harvard/d03e3925-d768-437b-91de-cc4bd6b94934
- https://trellis.stage.sinopia.io/repository/harvard/0e0f6e3c-166e-40ac-8293-19c9f2cb166f
- https://trellis.stage.sinopia.io/repository/penn/3a189a3e-a541-4612-b276-e23d2010e7e6
- https://trellis.stage.sinopia.io/repository/stanford/10a1f602-bc79-4271-a0f3-05c91e7ea613
- https://trellis.stage.sinopia.io/repository/stanford/43300a88-0e48-42f7-bc8c-45437bf9dcaf
- https://trellis.stage.sinopia.io/repository/stanford/764fdef1-5714-4b23-ae57-e483c353c62e
- h